File: 백준/Silver/20125. 쿠키의 신체 측정/쿠키의 신체 측정.py
import sys
from wsgiref.validate import header_re

input = sys.stdin.readline
N = int(input())
ls = [list(input().strip()) for _ in range(N)]
ans = [0,0,0,0,0] #왼쪽팔, 오른쪽팔, 허리, 왼쪽다리, 오른쪽다리
cookie = []
for i in range(N):
    for j in range(N):
        if ls[i][j] == '*':
            cookie.append([i+1, j+1])
heart = [cookie[0][0]+1, cookie[0][1]]
print(heart[0], heart[1])
cookie.pop(0)
for x, y in cookie:
    if x == heart[0] and y < heart[1]:
        ans[0] += 1
    elif x == heart[0] and y > heart[1]:
        ans[1] += 1
    elif x > heart[0] and y == heart[1]:
        ans[2] += 1
    elif x > heart[0] and y == heart[1]-1:
        ans[3] += 1
    elif x > heart[0] and y == heart[1]+1:
        ans[4] += 1
print(" ".join(map(str, ans)))

File: 백준/Silver/20125. 쿠키의 신체 측정/쿠키의 신체 측정.js
const fs = require("fs");
const { parse } = require("path");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const N = parseInt(input[0]);
const ls = input.slice(1, N + 1).map((line) => line.trim().split(""));
let cookie = [];
let ans = [0,0,0,0,0];

for (let i = 0; i < N; i++) {
  for (let j = 0; j < N; j++) {
    if (ls[i][j] === "*") {
      cookie.push([i + 1, j + 1]);
    }
  }
}
const heart = [cookie[0][0] + 1, cookie[0][1]];
console.log(heart[0], heart[1]);

for (const [x,y] of cookie){
  if (x === heart[0] && y < heart[1]){
    ans[0]++;
  }
  else if (x === heart[0] && y > heart[1]){
    ans[1]++;
  }
  else if (x > heart[0] && y === heart[1]){
    ans[2]++;
  }
  else if (x > heart[0] && y === heart[1]-1){
    ans[3]++;
  }
  else if (x > heart[0] && y === heart[1]+1){
    ans[4]++;
  }
}
console.log(ans.join(" "));
